dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/ClouderaClustersTask.java

```diff
@@ -0,0 +1,74 @@
+package com.google.edwmigration.dumper.application.dumper.connector.cloudera;
+
+import com.cloudera.api.swagger.ClustersResourceApi;
+import com.cloudera.api.swagger.client.ApiException;
+import com.cloudera.api.swagger.model.ApiCluster;
+import com.cloudera.api.swagger.model.ApiClusterList;
+import com.cloudera.api.swagger.model.ApiClusterUtilization;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.google.common.io.ByteSink;
+import com.google.edwmigration.dumper.application.dumper.handle.Handle;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+public class ClouderaClustersTask extends AbstractClouderaTask {
+
+    public ClouderaClustersTask() {
+        super("clusters.json");
+    }
+
+    @CheckForNull
+    @Override
+    protected Void doRun(TaskRunContext context, @Nonnull ByteSink sink, @Nonnull Handle handle)
+            throws Exception {
+        ClouderaHandle h = (ClouderaHandle) handle;
+        ApiClusterList apiClusterList = getClusters(h);
+        // TODO: Accept startDate as an input. Arbitrarily setting it to today minus 7
+        String startDate = LocalDateTime.now().minusDays(7).format(DateTimeFormatter.ISO_DATE);
+        try (Writer writer = sink.asCharSink(StandardCharsets.UTF_8).openBufferedStream()) {
+            for (ApiCluster apiCluster : apiClusterList.getItems()) {
+                //TODO: We should refactor this so the ClustersResourceApi object is just created once for both retrieving clusters and utilization
+                ApiClusterUtilization utilization = getClusterUtilization(h, apiCluster.getName(), startDate);
+                ClouderaClusterObject clouderaClusterObject = new ClouderaClusterObject();
+                clouderaClusterObject.setCluster(apiCluster);
+                clouderaClusterObject.setUtilization(utilization);
+                writer.write(ClouderaConnectorUtils.MAPPER.writeValueAsString(clouderaClusterObject));
```

Comment:
Can these use MAPPER.writeValue(writer, object) so we don't allocate String(s)?

Reply:
Question: Do we actually need to write as string because if the mapper throws an exception, we leave half a record in the file?